### PR TITLE
writing cast profiler results to disk

### PR DESCRIPTION
**Pull request by deyaaeldeen**

### CODE REVIEW

File: src/backend-c/generate-c.rkt [hunk 1]
```diff
@@ -140,7 +140,8 @@
 
 (define (print-cast-profile-result)
   (when (cast-profiler?)
-      (display "print_cast_profiler_results();")))
+    (display "print_cast_profiler_results();\n")
+    (display "write_cast_profiler_results_file(argv[0]);")))
```
Comment (akuhlens) [inline src/backend-c/generate-c.rkt]: I don't see any sort of error checking on this use of `argv`. What happens when the user forgets to pass an argument.

Comment (deyaaeldeen) [reply src/backend-c/generate-c.rkt]: It is not passed by the user, this memory location just represents the program name, according to the standard. The user supplied args starts from index 1.

Comment (akuhlens) [reply src/backend-c/generate-c.rkt]: Oh I see.
In that case the program ideally shouldn't write in the directory next to where the executable is located. Perhaps it should be a command line argument, or you can force it to show up in the directory where you run the program. Or we could just not worry about it for now and open a issue.

Comment (deyaaeldeen) [reply src/backend-c/generate-c.rkt]: I changed so that the file gets created in the current working directory. Done.

**Pull request merged**